src/server/services/wms/qgswmsrenderer.cpp

```diff
@@ -546,9 +547,34 @@ namespace QgsWms
       handlePrintErrors( layout.get() );
     }
 
+    // remove temeporary themes
+    // remove temeporary highlight layers
+    unConfigurePrintLayout( layout.get() );
```

Comment:
It seems that the `layout` parameter is a unique_ptr clone of the original layout coming from the project. So it'll be destroyed at the end of the `getPrint` method without adversely impacting the underlying project.

Am I wrong? Because in this case the `unConfigurePrintLayout` would be unnecessary.

Reply:
Oh, it seems that the project is not itself cloned in the `QgsPrintLayout::clone` method so you're right,  we have to post process  it.

But I still have one concern about the `unConfigurePrintLayout`. Actually, if something goes wrong during the `getPrint` method before reaching the `unConfigurePrintLayout`, the project is left in an uncleaned state (because the new theme and corresponding layers won't have been removed).

What do you think @jgrocha?

Reply:
@pblottiere If something goes wrong with `getPrint`, QGIS Server ends returning an exception or something else, but it will not use this highlight or layer configuration again. For the next request, QGIS Server will read the layout's layer configuration again from the project and the layers and highlight to print from the request.

In other words: is this `unConfigurePrintLayout` really necessary? If QGIS Server handles each request separately, if one request (like this one) changes the layer configuration (in memory), all other subsequent requests are not affected. 

To make sure, I've tested the same PR without `unConfigurePrintLayout`. All subsequent requests were not affected. But I'm probably missing something. Can you help me to find a situation in which a subsequent request can be affected by this change?

P.S. Thank you so much for taking a look into this.

Reply:
@pblottiere I'm so sorry, but inadvertently I've pushed my repository after rebasing. I had to recreate the pull request. This same PR is now #36007.